monai/data/image_writer.py

```diff
@@ -0,0 +1,395 @@
+        output_spatial_shape_ = output_spatial_shape_[:sp_dims]
+        original_channels = data.shape[3:]
+        if original_channels:  # multi channel, resampling each channel
+            data_np: np.ndarray = data.reshape(list(data.shape[:3]) + [-1])  # type: ignore
+            data_np = np.moveaxis(data_np, -1, 0)  # channel first for pytorch
+        else:  # single channel image, need to expand to have a channel
+            data_np = data[None]
+        affine_xform = AffineTransform(
+            normalized=False, mode=mode, padding_mode=padding_mode, align_corners=align_corners, reverse_indexing=True
+        )
+        data_torch = affine_xform(
+            torch.as_tensor(np.ascontiguousarray(data_np, dtype=dtype)).unsqueeze(0),
+            torch.as_tensor(np.ascontiguousarray(np.linalg.inv(_affine) @ target_affine, dtype=dtype)),
+            spatial_size=output_spatial_shape_,
+        )
+        data_np = data_torch[0].detach().cpu().numpy()
+        if original_channels:
+            data_np = np.moveaxis(data_np, 0, -1)  # channel last
+            data_np = data_np.reshape(list(data_np.shape[:3]) + list(original_channels))
+        else:
+            data_np = data_np[0]
+        return data_np, ensure_mat44(target_affine)
+
+    @classmethod
+    def convert_to_channel_last(
+        cls,
+        data: NdarrayOrTensor,
+        channel_dim: Optional[int] = 0,
+        squeeze_end_dims: bool = True,
+        spatial_ndim: Optional[int] = 3,
+        contiguous: bool = False,
+    ):
+        """
+        Rearrange the data array axes to make the `channel_dim`-th dim the last
+        dimension and ensure there are three spatial dimensions. If
+        ``channel_dim`` is ``None``, a new axis will be appended to the last
+        dimension.
+
+        When ``squeeze_end_dims`` is ``True``, a postprocessing step will be
+        applied to remove any trailing singleton dimensions.
+
+        Args:
+            data: input data to be converted to "channel-last" format.
+            channel_dim: specifies the axis of the data array that is the channel dimension.
+                ``None`` indicates no channel dimension.
+            squeeze_end_dims: if ``True``, any trailing singleton dimensions will be removed (after the channel
+                has been moved to the end). So if input is `(H,W,D,C)` and C==1, then it will be saved as `(H,W,D)`.
+                If D is also 1, it will be saved as `(H,W)`. If ``False``, image will always be saved as `(H,W,D,C)`.
+            spatial_ndim: modifying the spatial dims if needed, so that output to have at least
+                this number of spatial dims. If ``None``, the output will have the same number of
+                spatial dimensions as the input.
+            contiguous: if ``True``, the output will be contiguous.
+        """
+        # change data to "channel last" format
+        if channel_dim is not None:
+            # _chns = ensure_tuple_rep(channel_dim)
```

Comment:
please remove the comment code.

Thanks.

Reply:
Yes, I forgot to reenable the multiple channel_dim here, will update in the latest